### DISCUSSION TITLE
Character reading from console?

### DISCUSSION THREAD

**Discussion by AzuxirenLeadGuy** [mortbopet/Ripes#197] [Q&A] [answered]
Is reading characters (as input) from the console supported? If yes, please show an example of it.

**Comment by mortbopet** [comment, marked as answer]
Hi @AzuxirenLeadGuy

It is! Good point - there should probably be an example for this - 9ab12ac8cc6a66ec853b25939460c305388fd2b0 adds one.

**Reply by AzuxirenLeadGuy** [reply]
Great. Thanks a lot for this. Also, is there a way to read the characters as int or float (from this console input) as well, without manually converting it?

**Reply by mortbopet** [reply]
Not directly. For this you would need to write your own conversion function - which is actually rather trivial!
If you take a look at an ASCII table, you'll see that ASCII characters 0-9 have the char value of 48-57. So, you'd essentially just have to check that
1. the char `c` that you read is within this range and
2. return the value `c - 48` to get the integer that was read.

Hope this helps.

![image](https://user-images.githubusercontent.com/16338943/154452528-036022db-334d-4399-a9dc-dca6569e35dc.png)

**Reply by mortbopet** [reply]
Also worth exploring is that if you have the C compiler enabled and select the C example `consoleInput.c`, you'll see that we use `scanf` to read a floating point number. And, in the right-hand side of the editor you'll be able to see some of the C standard library code that actually implements that function call for you... which, is a lot of assembly! Hence why working at higher levels of abstraction starts to become appealing once you want your program to do less-than-trivial things.
